ocs_ci/utility/utils.py

```diff
@@ -397,27 +399,43 @@ def run_cmd(cmd, secrets=None, timeout=600, **kwargs):
         (str) Decoded stdout of command
 
     """
+    full_output = io.StringIO()
     masked_cmd = mask_secrets(cmd, secrets)
     log.info(f"Executing command: {masked_cmd}")
     if isinstance(cmd, str):
         cmd = shlex.split(cmd)
-    r = subprocess.run(
+    start_time = timeit.default_timer()
+    with subprocess.Popen(
         cmd,
         stdout=subprocess.PIPE,
         stderr=subprocess.PIPE,
-        stdin=subprocess.PIPE,
-        timeout=timeout,
+        universal_newlines=True,
         **kwargs
-    )
-    log.debug(f"Command output: {r.stdout.decode()}")
-    if r.stderr and not r.returncode:
-        log.warning(f"Command warning: {mask_secrets(r.stderr.decode(), secrets)}")
-    if r.returncode:
-        raise CommandFailed(
-            f"Error during execution of command: {masked_cmd}."
-            f"\nError is {mask_secrets(r.stderr.decode(), secrets)}"
-        )
-    return mask_secrets(r.stdout.decode(), secrets)
+    ) as p:
+        while True:
+            now = timeit.default_timer()
+            if (now - start_time) > timeout:
+                raise subprocess.TimeoutExpired(masked_cmd, timeout)
+            reads = [p.stdout.fileno(), p.stderr.fileno()]
+            ret = select.select(reads, [], [])
```

Comment:
@zmc since you raised original request with subprocess.Popen, do we need select here to achieve this?

Reply:
In case we want the live output of both stdout and stderr I guess we will need to use either select or poll. For now, all of my attempts to implement it with poll run into a deadlock, but if we agree we need live output of both pipes with poll, I can put some more time into it.

Reply:
@zmc ping? , I know we can do with select, but i guess then it can be achieved with subprocess.run as well.